toolkit/scripts/pkggen.mk

```diff
@@ -67,25 +67,34 @@ $(graph_file): $(specs_file) $(go-grapher)
 		$(logging_command) \
 		--output $@
 
+graphoptimizer_extra_flags :=
+
 # Remove any packages which don't need to be built, and flag any for rebuild if
 # their dependencies are updated.
 ifneq ($(CONFIG_FILE),)
 # If an optional config file is passed, validate it and any files it includes. The target should always depend
 # on the value of $(CONFIG_FILE) however, so keep $(depend_CONFIG_FILE) always.
 # Actual validation is handled in imggen.mk
+graphotimizer_extra_flags += --base-dir=$(CONFIG_BASE_DIR)
+
 $(optimized_file): $(validate-pkggen-config)
+
 endif
-$(optimized_file): $(graph_file) $(go-graphoptimizer) $(depend_PACKAGE_BUILD_LIST) $(depend_PACKAGE_REBUILD_LIST) $(depend_PACKAGE_IGNORE_LIST) $(toolchain_rpms) $(pkggen_rpms) $(CONFIG_FILE) $(depend_CONFIG_FILE)
+
+ifeq ($(REBUILD_DEPENDENT_PACKAGES), y)
+graphoptimizer_extra_flags += --rebuild-missing-dep-chains
+endif
+
+$(optimized_file): $(graph_file) $(go-graphoptimizer) $(depend_PACKAGE_BUILD_LIST) $(depend_PACKAGE_REBUILD_LIST) $(depend_PACKAGE_IGNORE_LIST) toolchain $(pkggen_rpms) $(CONFIG_FILE) $(depend_CONFIG_FILE)
```

Comment:
$(toolchain_rpms) are located outside of the `out` directory. Depending on them makes the build-packages target rebuild the toolchain packages - they need to be copied manually with copy-toolchain-files. We could depend on that target here and not rebuild toolchain packages, but the downside would be that they are copied every time you build packages.